resolver-dns/src/main/java/io/netty/resolver/dns/DnsAddressResolverGroup.java

```diff
@@ -46,20 +46,27 @@ public class DnsAddressResolverGroup extends AddressResolverGroup<InetSocketAddr
 
     public DnsAddressResolverGroup(DnsNameResolverBuilder dnsResolverBuilder) {
         this.dnsResolverBuilder = dnsResolverBuilder.copy();
+        this.dnsResolverBuilder
+                .resolveCache(dnsResolverBuilder.getOrNewCache()).cnameCache(dnsResolverBuilder.getOrNewCnameCache())
+                .authoritativeDnsServerCache(dnsResolverBuilder.getOrNewAuthoritativeDnsServerCache());
     }
```

Comment:
ffti: what you have is not incorrect but you could probably reduce some noise by adding a helper method that will perform all these cache setting operations for you which can be reused among the three different constructors. Maybe something like this:

```suggestion
    public DnsAddressResolverGroup(DnsNameResolverBuilder dnsResolverBuilder) {
        this.dnsResolverBuilder = withSharedCaches(dnsResolverBuilder.copy());
    }

    private static DnsNameResolverBuilder withSharedCaches(DnsNameResolverBuilder dnsResolverBuilder) {
        /// To avoid each member of the group having its own cache we either use the configured cache
        // or create a new one to share among the entire group.
        return dnsResolverBuilder.resolveCache(dnsResolverBuilder.getOrNewCache())
                .cnameCache(dnsResolverBuilder.getOrNewCnameCache())
                .authoritativeDnsServerCache(dnsResolverBuilder.getOrNewAuthoritativeDnsServerCache());
    }
```

Reply:
> ffti: what you have is not incorrect but you could probably reduce some noise by adding a helper method that will perform all these cache setting operations for you which can be reused among the three different constructors. Maybe something like this:

Sure, this version looks clearer in terms of semantics. Can I merge the suggested changes myself, or would you prefer to create a commit? It seems like this isn't quite finished yet.